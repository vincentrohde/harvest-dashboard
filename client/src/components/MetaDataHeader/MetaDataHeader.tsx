// Libs
import React from 'react';

// Styling
import './MetaDataHeader.scss';

interface MetaDataHeaderProps {
    entriesAmount: number;
    totalTime: number;
}

const MetaDataHeader = ({ entriesAmount, totalTime }: MetaDataHeaderProps) => (
    <div className="MetaDataHeader tab-container">
        <p className="meta-data pipes">
            <span>{ entriesAmount } Entries</span>
            <span>Total: { totalTime }</span>
        </p>
    </div>
);

export default MetaDataHeader;