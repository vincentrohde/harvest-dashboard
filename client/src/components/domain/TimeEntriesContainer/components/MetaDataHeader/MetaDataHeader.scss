.MetaDataHeader {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    .meta-data {
        flex: 1;
        text-align: center;
    }
}