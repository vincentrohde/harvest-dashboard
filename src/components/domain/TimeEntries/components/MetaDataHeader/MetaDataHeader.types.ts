export interface MetaDataHeaderProps {
    entriesAmount: number;
    totalHoursAndMinutes: string;
}
