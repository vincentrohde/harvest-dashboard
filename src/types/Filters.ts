export interface FiltersInterface {
    dateRange: string[] | 1;
}
