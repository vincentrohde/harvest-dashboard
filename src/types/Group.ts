export type groupsType = GroupInterface[];

export interface GroupInterface {
    id: number;
    name: string;
}