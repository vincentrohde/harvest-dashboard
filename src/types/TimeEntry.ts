import { ProjectInterface } from './Project';
import { TaskInterface } from './Task';

export type timeEntriesType = TimeEntryInterface[];

export interface ShallowTimeEntryInterface {
    hours: number | string;
    id: number;
    notes: string;
    spent_date: string;
    is_running: boolean;
    project: ProjectInterface;
    task: TaskInterface;
}

export interface TimeEntryInterface extends ShallowTimeEntryInterface {
    hours: number;
}

export interface ShallowSubmissionEntryInterface {
    notes: ShallowTimeEntryInterface['notes'];
    spent_date: ShallowTimeEntryInterface['spent_date'];
    hours: ShallowTimeEntryInterface['hours'];
    project_id: number | string;
    task_id: number | string;
}

export interface TimeEntrySubmissionInterface extends ShallowSubmissionEntryInterface {
    hours: TimeEntryInterface['hours'];
    project_id: number;
    task_id: number;
}