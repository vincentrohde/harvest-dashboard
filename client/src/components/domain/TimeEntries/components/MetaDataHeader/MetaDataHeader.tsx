// Libs
import React from 'react';

// Types
import { MetaDataHeaderProps } from './MetaDataHeader.types';

// Styling
import './MetaDataHeader.scss';

const MetaDataHeader = ({ entriesAmount, totalHoursAndMinutes }: MetaDataHeaderProps) => (
    <div className="MetaDataHeader tab-container">
        <p className="meta-data pipes">
            <span>{ entriesAmount } { entriesAmount > 1 ? 'Entries': 'Entry'}</span>
            <span>Total Time: { totalHoursAndMinutes }</span>
        </p>
    </div>
);

export default MetaDataHeader;
