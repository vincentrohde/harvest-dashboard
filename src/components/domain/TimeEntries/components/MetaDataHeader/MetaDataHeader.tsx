// Libs
import React from 'react';
import classNames from 'classnames';

// Types
import { MetaDataHeaderProps } from './MetaDataHeader.types';

// Styling
import styles from './MetaDataHeader.module.scss';

const MetaDataHeader = ({ entriesAmount, totalHoursAndMinutes }: MetaDataHeaderProps) => (
    <div className={classNames(styles.MetaDataHeader, 'tab-container')}>
        <p className={classNames(styles['meta-data'], 'pipes')}>
            <span>{ entriesAmount } { entriesAmount > 1 ? 'Entries': 'Entry'}</span>
            <span>Total Time: { totalHoursAndMinutes }</span>
        </p>
    </div>
);

export default MetaDataHeader;
