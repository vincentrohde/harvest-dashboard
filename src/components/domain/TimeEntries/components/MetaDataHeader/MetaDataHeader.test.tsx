// Libs
import { render } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom/extend-expect';

// Components
import MetaDataHeader from './MetaDataHeader';

// Types
import { MetaDataHeaderProps } from './MetaDataHeader.types';

const mockData: MetaDataHeaderProps = {
    entriesAmount: 20,
    totalHoursAndMinutes: '2:00'
}

describe('<MetaDataHeader />', () => {
    it('renders', () => {
        const { container } = render(
            <MetaDataHeader {...mockData} />
        );

        expect(container).toMatchSnapshot();
    });
})
