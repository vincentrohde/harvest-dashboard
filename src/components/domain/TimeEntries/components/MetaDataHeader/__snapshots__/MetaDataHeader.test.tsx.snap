// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaDataHeader /> renders 1`] = `
<DocumentFragment>
  <div
    class="MetaDataHeader tab-container"
  >
    <p
      class="meta-data pipes"
    >
      <span>
        20 Entries
      </span>
      <span>
        Total Time: 2:00
      </span>
    </p>
  </div>
</DocumentFragment>
`;
